[3.0] Change likes to reactions...

Oldiesmann opened the pull request:
This PR changes SMF's likes feature into a full-fledged reactions system.

Features:
Admins can add/remove reactions as desired (only limit is defined by the limits of the SMALLINT data type, which should be plenty)
For simplicity, it will use images in a reactions folder in the theme's images folder and will look for a $txt['react_{name}'] string for i18n (defaults to 'name' defined in the database if not found)

Work done so far:
- Database changes for install and upgrade
- Modified language strings

Still to do:
- Code modifications/additions to handle things
- Theme stuff
- Testing (this is alpha so not too worried about that yet)

Note that the upgrade uses a combination of db API functions and upgrade_query. In some situations (such as creating a table), upgrade_query is far simpler than the associated db API call.

Oldiesmann committed:
Initial commit to change likes to reactions...

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Create the reactions table and change the ID column to match SMF standard naming.

Oldiesmann committed:
Handle upgrades - creating/renaming things as needed.

Oldiesmann committed:
Consistency is a good thing...

Oldiesmann commented:
Also cc @Sesquipedalian since this will require changes to the DB schema stuff in #8093

Oldiesmann committed:
Update with changes from recurring events

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Update permission and setting...

Oldiesmann committed:
Rename Like.php and the Like class. Also adjusted another language string

Oldiesmann committed:
More renaming. Also some other minor fixes

Oldiesmann committed:
Missed something

Oldiesmann committed:
Pull in latest upstream changes

asmith20002 commented:
Good job. IMO this also should be done for 2.1

live627 commented:
2.1 is feature frozen and receives bug fixes only.

On Sun, May 5, 2024 at 9:52 PM asmith20002 ***@***.***> wrote:

> Good job. IMO this also should be done for 2.1
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SimpleMachines/SMF/pull/8187#issuecomment-2095194970>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADJNN5J44LLS3XBZI3OVHLZA4EAPAVCNFSM6AAAAABG3Y3IX6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDAOJVGE4TIOJXGA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions
latest upstream changes

Oldiesmann committed:
Merge latest upstream changes

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Add support for changing the order of available reactions

Oldiesmann committed:
Change getLikedMsgs to getReactedMsgs and update the logic slightly

Oldiesmann committed:
Change a few more things...

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Fix a typo and update the stats info.

Oldiesmann committed:
Language file tweak for stats.

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Rename the template and a bunch of other stuff...

Oldiesmann committed:
Create a new trait for loading reactions.

Oldiesmann committed:
This is handled by the trait so don't need it now...

Oldiesmann committed:
A couple of minor tweaks...

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Add the initial file for the admin stuff and modify the admin menu

Oldiesmann committed:
Move the settings to the new file and add initial code for handling updates/deletions/additions

Oldiesmann committed:
Minor tweaks and such

Oldiesmann committed:
Fix the typo, add a comment block and remove the unneeded trait property

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Code to handle adding/removing/updating reactions and initial code for the list. More to come...

Oldiesmann committed:
HTML5, not XHTML...

Oldiesmann committed:
More random fixes and such

Oldiesmann committed:
A few more minor tweaks:
- Add code to handle separate buttons for deleting/saving
- Add placeholders for additional row and inline JS
- Add closing PHP tag for consistency

Oldiesmann committed:
More fun...
- Update the get_items function per specs
- Move the do_update conditional to the proper spot
- Add the buttons and a checkbox column for deleting stuff

DiegoAndresCortes commented on Sources/Actions/Admin/Reactions.php:
```diff
@@ -285,6 +287,39 @@ public function editreactions(): void
+				// Clicking this magic link adds a new row...
+				'position' => 'bottom_of_list',
+				'data' => [
+					'value' => '<a href="javascript:void(0);" onclick="addrow()">' . Lang::$txt['reacts_add'] . '</a>'
```
Can you make this a <button type="button" instead so we don't have to come back to it when fixing for wcag sutff

Oldiesmann replied:
Sure. I'm just writing up basic code at this point and haven't even tested any of this yet (I plan to start that soon so I don't end up with a ton of stuff to fix later)

Oldiesmann committed:
- Change new row thing from link to button
- Add javascript for adding said new row

Oldiesmann committed:
Add function to output the data for the check column

Oldiesmann committed:
Add constructor, change the likes permission text string and create language file.

Oldiesmann committed:
Finish menu and add language strings

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Consistency and all that...

Oldiesmann committed:
Oops

Oldiesmann committed:
Another typo; also fix typos in Postgres

Oldiesmann committed:
Another typo and work around upgrade_query not supporting {db_prefix}

Oldiesmann committed:
Fun fact: PHP doesn't like trying to implode() a string

Oldiesmann committed:
A few more minor fixes

Oldiesmann committed:
More fixes and general cleanup.

Oldiesmann committed:
Missing language strings

Oldiesmann committed:
There's such a thing as too many permissions, right?

Oldiesmann committed:
Forgot to finish the settings stuff...

Oldiesmann committed:
More missing language strings

Oldiesmann committed:
A couple more typos

Oldiesmann committed:
More bugs

Oldiesmann committed:
More typos and missing language strings

Oldiesmann committed:
Typos are sooo much fun...

DiegoAndresCortes commented on other/install_3-0_MySQL.sql:
```diff
@@ -991,6 +991,17 @@ CREATE TABLE {$db_prefix}qanda (
+CREATE TABLE {$db_prefix}reactions (
+	id_reaction SMALLINT UNSIGNED AUTO_INCREMENT,
+	name VARCHAR(255) NOT NULL DEFAULT '',
+	order SMALLINT UNSIGNED NOT NULL DEFAULT '0',
```
```suggestion
	`order` SMALLINT UNSIGNED NOT NULL DEFAULT '0',
```

Oldiesmann replied:
I dropped that column entirely for the time being. I was considering adding functionality to choose the order of the reactions but forgot about that when I wrote all the admin code and didn't want to go to the trouble of adding that in as well. I'll do that if I add it back at some point though.

Oldiesmann committed:
More cleanup

Oldiesmann committed:
Remove currently unused column from install SQL files

Oldiesmann committed:
Fix URL for saving settings

Oldiesmann committed:
Found a missing quote

Oldiesmann committed:
Duplicate column

Oldiesmann committed:
Typo

Oldiesmann committed:
A few more minor tweaks

Oldiesmann committed:
Another typo

Oldiesmann committed:
Fix our add button

Oldiesmann committed:
A couple more typos

Oldiesmann committed:
Make our add button look nice

Oldiesmann committed:
Some more tweaks/fixes

Oldiesmann committed:
Another typo

Oldiesmann committed:
Validate token while saving settings

Oldiesmann committed:
Add a title to the settings page and a token to the form on the manage page

Oldiesmann committed:
Merge branch 'SimpleMachines:release-3.0' into reactions

Oldiesmann committed:
Typo

Oldiesmann committed:
Merge branch 'reactions' of https://github.com/Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'release-3.0' into reactions

Oldiesmann committed:
Fix admin menu stuff.

Oldiesmann committed:
Token name typo

Oldiesmann committed:
Add a key to the insert because the code has a meltdown otherwise...

Oldiesmann committed:
Fix the queries

Oldiesmann committed:
Fix the queries the right way this time

Oldiesmann committed:
Another typo

Oldiesmann committed:
Missed those somehow...

Oldiesmann committed:
GROUP BY goes after WHERE, not before

Oldiesmann committed:
Too many typos...

Oldiesmann committed:
Fix our updates array

Oldiesmann committed:
Properly fix the update this time.

Oldiesmann committed:
Final fixes. Everything will work now.

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Add code to load reaction information for posts

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Likes -> reacts...

Oldiesmann committed:
Delete reactions when a topic is deleted

Oldiesmann committed:
Make sure the reaction ID passed in the URL is valid

Oldiesmann committed:
Merge branch 'smf3' of https://github.com/Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge latest changes from upstream

Oldiesmann committed:
Apparently I missed a couple changes a while back...

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'smf3' of https://github.com/Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'reactions' of https://github.com/Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Forgotten commas will kill you...

Oldiesmann committed:
Somehow the section to manage reactions settings went missing. I swear it was there at one time...

Oldiesmann committed:
Likes are a thing of the past...

Oldiesmann committed:
Update the version to match the rest.

Oldiesmann committed:
Fix call to undefined method

Oldiesmann committed:
Move a few language strings so the menu items show up properly

Oldiesmann committed:
Merge branch 'reactions' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of https://github.com/SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'reactions' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of https://github.com/SimpleMachines/SMF into reactions

Oldiesmann committed:
Replace "getLikedMsgs" calls with "getReactedMsgs"

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Merge branch 'reactions' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'smf3' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of https://github.com/SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of https://github.com/SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of https://github.com/SimpleMachines/SMF into reactions

Sesquipedalian commented:
This is still marked as a draft, @Oldiesmann. Is it ready for review?

Oldiesmann commented:
Not yet. The back end stuff is done but I haven't done anything on the user side yet because theme-related stuff isn't my strong point. It's at a point now where the admin can add, remove and modify reactions as well as choose who can react to things, but I don't have any of the UI stuff done - the part that actually allows someone to react to something.

jdarwood007 commented:
I would suggest putting something in that looks decent and letting the theme PR take care of improving it then.

Oldiesmann commented:
> I would suggest putting something in that looks decent and letting the theme PR take care of improving it then.

@tyrsson is going to help me out with the UI part when he gets a chance.

tyrsson commented:
> > I would suggest putting something in that looks decent and letting the theme PR take care of improving it then.
> 
> @tyrsson is going to help me out with the UI part when he gets a chance.

I hope to get started on it in the next couple weeks. It's on the schedule. The markup really is not the part that will take some doing it will be the JS required to make it really usable that needs sorting out.

Oldiesmann commented:
@DiegoAndresCortes did you mean to request changes? If so, what do you suggest? GitHub only says that you "requested changes with read-only permissions"

DiegoAndresCortes commented:
No recently, only the two older ones where you commented

Oldiesmann committed:
Merge branch 'release-3.0' of https://github.com/SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Pull in latest changes and adapt my changes to new install/upgrade structure

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann commented:
Going to go ahead and mark this as ready for review. At the very least we can merge the initial backend functionality in, which will allow others to contribute without having to fork my branch.

This currently has all the functionality to allow admins to manage reactions - enable/disable them, choose who can use them and add/edit/delete them. The user-facing stuff doesn't work yet as I'm not good with UI stuff at all. I have designed it with i18n in mind as well. The plan is that, in the UI area, we can use `$txt['react_{name}']` for the reaction name, or just fall back to the name from the DB if no such text string exists.

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

tyrsson commented:
Sorry @Oldiesmann I have just not been able to get to this :( As you know from our Discord convo's I've got a lot of irons in the fire at the moment.

jdarwood007 commented on Sources/Maintenance/Migration/v3_0/HolidaysToEvents.php:
```diff
@@ -679,4 +679,4 @@ public function execute(): bool
 		return true;
 	}
-}
+}
```
Revert this change

Oldiesmann replied:
I'm not sure what specifically I need to revert here. It looks like I just removed a } and added one back?

This file got messed up when I pulled in the changes from the install/upgrade stuff, Git was marking the file as "changed in both" even though it was new to my branch and I didn't touch it. I ended up copying the raw file from GitHub and replacing my version with that one because I couldn't figure out how to resolve the conflict otherwise.

jdarwood007 replied:
I think its the line ending that changed.

Oldiesmann replied:
I'm not sure what I can do to fix that but I'll see.

Oldiesmann replied:
Replaced it with the raw file from GitHub. Looks like it was an empty line at the end of the file that was missing, but this should be fixed now

jdarwood007 commented on Sources/Maintenance/Migration/v3_0/UserReactions.php:
```diff
@@ -0,0 +1,161 @@
+	/****************
+	 * Public methods
+	 ****************/
+
```
```suggestion
	public function isCandidate(): bool
	{
		// Make sure we haven't already done this...
		$cols = Db::$db->list_columns('messages');

		// If the reactions column exists in the messages table, there's nothing to do
		return !in_array('reactions', $cols))
    }

```

If you add this check, you can remove it from the execute() function.  The upgrader will indicate "skip" when it comes across it and isCandidate() returns false.

jdarwood007 commented on Sources/Maintenance/Migration/v3_0/UserReactions.php:
```diff
@@ -0,0 +1,161 @@
+			if (!empty($table_exists))
+			{
+				// Rename the table
+				Db::$db->query('', 'ALTER TABLE {db_prefix}user_likes RENAME TO {db_prefix}user_reacts');
```
@Sesquipedalian 
This isn't cross DB safe, but we don't have a method to rename a table.  It seems like we should have such logic.

Sesquipedalian replied:
Yes, we should.

Sesquipedalian replied:
Added in #8828

jdarwood007 commented on Sources/Maintenance/Migration/v3_0/UserReactions.php:
```diff
@@ -0,0 +1,161 @@
+				Db::$db->query('', 'ALTER TABLE {db_prefix}user_likes RENAME TO {db_prefix}user_reacts');
+				
+				// Add the new column
+				Db::$db->add_column('{db_prefix}user_reacts', ['name' => 'id_reaction', 'type' => 'smallint', 'not_null' => true, 'default' => '0']);
```
Ensure that you have a use statement to `UserReacts` and you could do:
```
$tbl = new UserReacts();

Db::$db->add_column('{db_prefix}user_reacts', $tbl->columns['id_reaction']);
```

Thus ensuring that you are asking it to do the changes as needed to match the schema.

jdarwood007 commented on Sources/Maintenance/Migration/v3_0/UserReactions.php:
```diff
@@ -0,0 +1,161 @@
+				Db::$db->change_column('{db_prefix}user_reacts', 'like_time', ['name' => 'react_time']);
+
+				// Rename the index
+				Db::$db->query('', 'ALTER TABLE {db_prefix}user_reacts RENAME idx_liker TO idx_reactor');
```
Use `Db::$db->rename_index`

jdarwood007 commented on Sources/Maintenance/Migration/v3_0/UserReactions.php:
```diff
@@ -0,0 +1,161 @@
+				
+				if(isset($idx->type))
+					$indexes[$idx->name]['type'] = $idx->type;
+			}
+
+			// Now we just create the table
+			Db::$db->create_table('{db_prefix}user_reacts', $columns, $indexes);
+
+			// Add the reactions column and related index to the messages table
+			Db::$db->add_column('{db_prefix}messages', ['name' => 'reactions', 'type' => 'smallint', 'not_null' => true, 'default' => '0']);
+			Db::$db->add_index('{db_prefix}messages', ['name' => 'idx_messages_reactions', 'columns' => ['reactions']]);
+		}
+
+		// Create the reactions table. Only two columns here so no need to save space by looping through the schema
+		$reactions_cols = [
+			'id_reaction' => [
+				'name' => 'id_reaction',
+				'type' => 'smallint',
+				'unsigned' => true,
+				'not_null' => true,
+				'default' => 0,
+			],
+			'name' => [
+				'name' => 'name',
+				'type' => 'varchar',
+				'size' => '255',
+				'not_null' => true,
+				'default' => '',
+			]
+		];
+
+		$reactions_idxs = [
+			'id_reaction' => [
+				'name' => 'id_reaction',
+				'columns' => ['id_reaction'],
+				'type' => 'primary'
+			]
+		];
+
+		Db::$db->create_table('{db_prefix}reactions', $reactions_cols, $reactions_idxs);
```
```suggestion
			$table = new Schema\v3_0\UserReacts();
			$table->create();
```

The upgrader will handle all the logic to setting up the table as long as the schema has all the correct defs.

Also you could change your entire check logic at the start to do if statement this way:
```
			$existing_tables = Db::$db->list_tables();

			if (!in_array(Config::$db_prefix . $table->name, $existing_tables)) {
				$table->create();
			} else {
			[....]
```

jdarwood007 reviewed (commented):
Just some upgrader comments

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Use the built-in index renaming method

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Use the soon-to-be-implemented rename_table method

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Use column definition directly from schema

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Refactor/simplify things a bit

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Attempt to fix HolidaysToEvents file

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Yay for typos...

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Identifier parameter no longer used...

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:Simplemachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Clean up stray reference to likes

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:Simplemachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'reactions' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Remove likes area in admin and fix admin search

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
The old identifier parameter doesn't exist anymore...

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Fix some more queries

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Missed one

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Update query to match actual column name...

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann committed:
Missed a "reacts" reference

Signed-off-by: Michael Eshom <oldiesmann@gmail.com>

Oldiesmann commented:
Fixed several bugs:
- Fixed admin center - fixed admin search referencing non-existent function; removed likes; added reactions menu item
- Fixed queries that still used the old format which had an optional identifier as the first parameter
- Fixed query in Stats that referenced a non-existent "reacts" column

Still need to update the Msg.php code to populate the variable which tracks how many of each reaction was chosen for a particular post

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

tyrsson commented on Languages/en_US/Profile.php:
```diff
@@ -136,7 +136,7 @@
 $txt['alert_msg_mention'] = 'When my @name is mentioned in a post';
 $txt['alert_msg_quote'] = 'When one of my posts is quoted';
-$txt['alert_msg_like'] = 'When one of my posts is liked';
+$txt['alert_msg_react'] = 'When someoe reacts to one of my posts';
```
Got a typo in the string. "someoe"

Oldiesmann committed:
Squash a typo

Signed-off-by: oldiesmann <oldiesmann@gmail.com>

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Signed-off-by: oldiesmann <oldiesmann@gmail.com>

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Fix missing {

Signed-off-by: oldiesmann <oldiesmann@gmail.com>

Oldiesmann commented:
@jdarwood007 @Sesquipedalian Can either of you explain what's causing the PHP-CS-Fixer check to fail here? It's not showing me an error and I can't get it to run locally (it just outputs whatever HTML is generated by SMF)

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Code cleanup. Attempt to make php-cs-fixer happy.

Signed-off-by: oldiesmann <oldiesmann@gmail.com>

Oldiesmann committed:
More cleanup

Signed-off-by: oldiesmann <oldiesmann@gmail.com>

Oldiesmann committed:
Even more cleanup...

Signed-off-by: oldiesmann <oldiesmann@gmail.com>

Oldiesmann committed:
Attempting to fix the newline issue

Signed-off-by: oldiesmann <oldiesmann@gmail.com>

Sesquipedalian commented:
I see you are working to fix code style issue, @Oldiesmann. I recommend that you simply [install PHP-CS-Fixer](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer?tab=readme-ov-file#installation) and then use it to fix your files, like so:

```
php-cs-fixer fix --config=".php-cs-fixer.dist.php" --allow-risky="yes"
```

The `--allow-risky="yes"` bit is needed because we do use one rule that PHP-CS-Fixer classifies as "risky."

Oldiesmann committed:
Fix the remaining issues

Signed-off-by: oldiesmann <oldiesmann@gmail.com>

Oldiesmann commented:
> I see you are working to fix code style issue, @Oldiesmann. I recommend that you simply [install PHP-CS-Fixer](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer?tab=readme-ov-file#installation) and then use it to fix your files, like so:
> 
> ```
> php-cs-fixer fix --config=".php-cs-fixer.dist.php" --allow-risky="yes"
> ```

It doesn't work if I run it locally because of all the index.php files SMF sticks within vendor and its various subdirectories - it just ends up pulling in SMF's own index.php file and generates whatever output is generated from that.

Sesquipedalian commented:
Have you tried it recently? The .php-cs-fixer.dist.php file is configured to ignore anything inside the vendor dir.

Oldiesmann commented:
If I run it from the directory with the now-fixed files, it complains that Settings.php isn't readable, and if I copy Settings.php to the directory, it'll just spit out the HTML for the error message saying SMF can't connect to the database. If I run it from a live local test board, it spits out HTML.

<img width="1705" height="154" alt="image" src="https://github.com/user-attachments/assets/864e8bc8-86ad-4704-b4ce-9a858c8e51b1" />

<img width="1708" height="361" alt="image" src="https://github.com/user-attachments/assets/4d385809-ecfc-49fc-bd94-2be85038d57e" />

Sesquipedalian commented:
Hm. You are correct. I will fix this soon.

Sesquipedalian commented:
Please see #9114 for the fix to your problem, @Oldiesmann.

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'reactions' of github.com:Oldiesmann/SMF21 into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Signed-off-by: oldiesmann <oldiesmann@gmail.com>

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann committed:
Merge branch 'release-3.0' of github.com:SimpleMachines/SMF into reactions

Oldiesmann commented:
> Is any of the TODO stuff still relevant?
> 

Most of it is. I've added the functionality on the admin side (add/edit/remove/reorder reactions, enable/disable the feature and choose who can react) but none of the user side functionality is there there yet. @tyrsson is/was going to handle most of that at some point.
